packages/atlas/src/views/viewer/MarketplaceView/FeaturedNftsSection/FeatureNftModal.tsx

```diff
@@ -0,0 +1,202 @@
+      })
+      handleClose()
+    },
+  })
+
+  const { displaySnackbar } = useSnackbar()
+
+  const inputRef = useRef<HTMLInputElement | null>(null)
+
+  const submit = () => {
+    handleSubmit(async (data) => {
+      const videoId = /[^/]*$/.exec(data.url)?.[0]
+      if (!videoId) {
+        return
+      }
+
+      await requestNftFeaturedMutation({
+        variables: {
+          nftId: videoId,
+          rationale: '',
+        },
+      })
+    })()
+  }
+
+  const client = useApolloClient()
+
+  useEffect(() => {
+    if (isOpen) {
+      inputRef.current?.focus()
+    }
+  }, [isOpen])
+
+  const validateNft = useCallback(
+    async (id?: string) => {
+      if (!id) {
+        return 'Enter a valid link to your video NFT.'
+      }
+      const {
+        data: { ownedNftById },
+      } = await client.query<GetNftQuery, GetNftQueryVariables>({
+        query: GetNftDocument,
+        variables: {
+          id: id,
+        },
+      })
+      if (!ownedNftById) {
+        return 'This video is not an NFT.'
+      }
+      setVideoId(id)
+      if (
+        ownedNftById.transactionalStatus?.__typename === 'TransactionalStatusIdle' ||
+        ownedNftById.transactionalStatus?.__typename === 'TransactionalStatusInitiatedOfferToMember'
+      ) {
+        return "This video's NFT is not put on sale."
+      }
+
+      if (
+        ownedNftById.transactionalStatus?.__typename === 'TransactionalStatusAuction' &&
+        ownedNftById.transactionalStatus.auction.isCompleted
+      ) {
+        return "This video's NFT is not put on sale."
+      }
+      return true
+    },
+    [client]
+  )
+
+  const { ref, ...inputRefRest } = useMemo(() => {
+    return register('url', {
+      onChange: debouncePromise(
+        async () => {
+          await trigger('url')
+          setIsInputValidating(false)
+        },
+        500,
+        {
+          key() {
+            setIsInputValidating(true)
+            return null
+          },
+        }
+      ),
+      required: { value: true, message: 'Enter link to your video NFT.' },
+      validate: {
+        validUrl: (val) => {
+          return val.startsWith(window.location.origin + '/video/') ? true : 'Enter a valid link to your video NFT.'
+        },
+        nftIsValid: async (val) => {
+          // get the last string after slash
+          const videoId = /[^/]*$/.exec(val)?.[0]
+          const validation = await validateNft(videoId)
+
+          return validation
+        },
+      },
+    })
+  }, [register, trigger, validateNft])
+
+  const isInputRefActiveElement = document.activeElement === inputRef.current
+  return (
+    <DialogModal
+      title="Apply for featured section"
+      show={isOpen}
+      noContentPadding
+      primaryButton={{
+        text: loading ? 'Please wait...' : 'Submit video',
+        disabled: loading,
+        onClick: submit,
+      }}
+      secondaryButton={{
+        text: 'Cancel',
+        disabled: loading,
```

Comment:
Should we disable Cancel button when mutation is loading?

Reply:
To elaborate - what I mean is that when connection gets broken somehow, user will end up locked in the dialog?